container/prune_tar.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2015 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This tool build tar files from a container_layer,
+while excluding some files and directories."""
+
+import gflags
+import json
+import sys
+
+from container import archive
+
+gflags.DEFINE_string('output', None, 'The output file, mandatory')
+gflags.MarkFlagAsRequired('output')
+
+gflags.DEFINE_string(
+    'manifest', None, 'JSON manifest of contents to remove from the layer')
+
+gflags.DEFINE_string(
+    'root_directory', '.', 'Default root directory is named "."'
+    'Windows docker images require this be named "Files" instead of "."')
+
+FLAGS = gflags.FLAGS
+
+
+class TarFile(object):
+  """A class to generate a pruned container layer."""
+
+  def __init__(self, output, root_directory, tar, remove_paths):
+    self.output = output
+    self.root_directory = root_directory
+    self.tar = tar
+    self.remove_paths = remove_paths
+
+  def __enter__(self):
+    self.tarfile = archive.TarFileWriter(self.output)
+    return self
+
+  def __exit__(self, t, v, traceback):
+    self.tarfile.close()
+
+  def name_filter(self, path):
+    """Callback for archive.TarFileWriter.add_tar that returns false
+    if the file or directory in remove_paths.
```

Comment:
Suggest changing this comment to describe the scenario when the function returns false, so something like:

returns false when the given path should be removed from the layer.